content/en/contributing/development-environment-setup.md

```diff
@@ -7,69 +7,101 @@ aliases:
 
-{{< command >}}
-$ make start
-{{< / command >}}
+#### Python Dependencies
 
-### Building the Docker image for Development
+* [JPype1](https://pypi.org/project/JPype1/) might require `g++` to fix a compile error on ARM Linux `gcc: fatal error: cannot execute ‘cc1plus’`
+  * Used in EventBridge, EventBridge Pipes, and Lambda Event Source Mapping for a Java-based event ruler via the opt-in configuration `EVENT_RULE_ENGINE=java`
+  * Introduced in [#10615](https://github.com/localstack/localstack/pull/10615)
+* [libvirt-python](https://pypi.org/project/libvirt-python/) requires `libvirt-dev` on Debian or `libvirt` on macOS/Brew to fix `Package libvirt was not found in the pkg-config search path.`
+  * Used in EC2 to access Libvirt inside the LocaStack container
+  * Introduced in [localstack-ext#2827](https://github.com/localstack/localstack-ext/pull/2827)
 
-Please note that there are a few commands we need to run on the host to prepare the local environment for the Docker build - specifically, downloading some dependencies like the StepFunctions local binary. Therefore, simply running `docker build .` in a fresh clone of the repo may not work.
+#### Uncategorized
 
-We generally recommend using this command to build the Docker image locally (works on Linux/MacOS):
+Some services or tests might need some of these dependencies (not yet categorized):
 
-{{< command >}}
-$ make docker-build
-{{< / command >}}
+* [OpenJDK](https://openjdk.org/install/)
+* [NodeJS & npm](https://nodejs.org/en/download/)
+* [Maven](https://maven.apache.org/download.cgi)
+* [Gradle](https://gradle.org/install/)
+* [Terraform](https://www.terraform.io/downloads)
+
+#### Lambda
+
+* macOS users need to configure `LAMBDA_DEV_PORT_EXPOSE=1` such that the host can reach Lambda containers via IPv4 in bridge mode (see [#7367](https://github.com/localstack/localstack/pull/7367)).  
+
+#### EVENT_RULE_ENGINE=java
+
+* Requires Java to execute to invoke the AWS [event-ruler](https://github.com/aws/event-ruler) using [JPype](https://github.com/jpype-project/jpype), a Python to Java bridge.
+* Set `JAVA_HOME` to a JDK installation. For example: `JAVA_HOME=/opt/homebrew/Cellar/openjdk/21.0.2`
+
+### Changing our fork of moto
+
+1. Fork our moto repository on GitHub [https://github.com/localstack/moto](https://github.com/localstack/moto)
+2. Clone the forked moto repository `git clone git@github.com:<GITHUB_USERNAME>/moto.git` (using the `localstack` branch)
+3. Within the localstack repository, install moto in **editable** mode:
+
+```sh
+# Assuming the following directory structure:
+#.
+#├── localstack
+#└── moto
+
+cd localstack
+source .venv/bin/activate
+
+pip install -e ../moto
+```
 
 ### Tips
 
+* Most of the contributors use the free community version of [PyCharm](https://www.jetbrains.com/pycharm/).
```

Comment:
Well, is it necessary to state an opinion on the usage of the IDE? This shouldn't make any difference, does it?

Reply:
removed
(it was my attempt to hide it here, given it was the prominent step 3. of the setup instructions before 😄 )